packages/core/src/state-management/state-manager.ts

```diff
@@ -442,17 +454,28 @@ export class StateManager {
             }
             case AnchorRequestStatusName.COMPLETED: {
               await this._handleAnchorCommit(state$, asr.cid, asr.anchorCommit.cid, asr.witnessCar)
-              await this.anchorRequestStore.remove(state$.id)
+
+              if (!this._tipHasProcessingAnchorRequest(state$)) {
+                await this.anchorRequestStore.remove(state$.id)
+              }
               stopSignal.next()
               return
             }
             case AnchorRequestStatusName.FAILED: {
-              if (!asr.cid.equals(state$.tip)) return
               this.logger.warn(
                 `Anchor failed for commit ${asr.cid} of stream ${asr.streamId}: ${asr.message}`
               )
-              state$.next({ ...state$.value, anchorStatus: AnchorStatus.FAILED })
-              await this.anchorRequestStore.remove(state$.id)
+
+              // if this is the anchor response for the tip update the state
+              if (asr.cid.equals(state$.tip)) {
+                state$.next({ ...state$.value, anchorStatus: AnchorStatus.FAILED })
+              }
+
+              if (!this._tipHasProcessingAnchorRequest(state$)) {
+                await this.anchorRequestStore.remove(state$.id)
```

Comment:
do we also need to remove from the `anchorRequestStore` for the REPLACED status below?

Reply:
I think we can safely assume if we received a REPLACED status, that means that we have another outstanding request for a commit for that stream

Reply:
I can also add an extra check as above. It seems based on https://linear.app/3boxlabs/issue/CDB-2622/investigate-why-ceramic-node-is-polling-for-very-old-anchor-request we can get into very weird states.